### PR TITLE
Implement postgresql-10 support

### PR DESCRIPTION
**Pull request by CyberDem0n**
* rename pg_xlog => pg_wal
* small changes in bgworker api
* changes in pg_stat_activity (now it shows all processes, not only backends)

### CODE REVIEW

File: disk_stats.c [hunk 1]
```diff
@@ -27,7 +27,11 @@ typedef struct {
 	bool me_remote;
 } mount_entry;
 
+#if PG_VERSION_NUM >= 100000
+static const char pg_wal[] = "pg_wal";
```
Comment (avaczi) [inline disk_stats.c]: indentation?

File: bg_mon.c [hunk 2]
```diff
@@ -402,11 +416,17 @@ _PG_init(void)
 		return;
 
 	/* set up common data for all our workers */
+        memset(&worker, 0, sizeof(worker));
 	worker.bgw_flags = BGWORKER_SHMEM_ACCESS |
 		BGWORKER_BACKEND_DATABASE_CONNECTION;
 	worker.bgw_start_time = BgWorkerStart_ConsistentState;
 	worker.bgw_restart_time = 1;
+#if PG_VERSION_NUM >= 100000
+        sprintf(worker.bgw_library_name, "bg_mon");
```
Comment (a1exsh) [inline bg_mon.c]: Why not strcpy?

File: bg_mon.c [hunk 3]
```diff
@@ -402,11 +416,17 @@ _PG_init(void)
 		return;
 
 	/* set up common data for all our workers */
+        memset(&worker, 0, sizeof(worker));
```
Comment (a1exsh) [inline bg_mon.c]: Looks like a useful thing anyway, but has just one extra leading space.

File: postgres_stats.c [hunk 4]
```diff
@@ -67,6 +67,7 @@ static const char * const pg_stat_activity_query =
 							"AND other.granted = 't' "
 #endif
 "WHERE a.pid != pg_backend_pid() "
+  "AND datname IS NOT NULL "
```
Comment (a1exsh) [inline postgres_stats.c:70]: Is this also due to v10?

File: bg_mon.c (hunk 2 above)
Comment (CyberDem0n) [reply bg_mon.c]: Because copy&paste from src/test/modules/worker_spi/worker_spi.c

File: bg_mon.c (hunk 3 above)
Comment (CyberDem0n) [reply bg_mon.c]: It also came from src/test/modules/worker_spi/worker_spi.c

File: postgres_stats.c (hunk 4 above)
Comment (CyberDem0n) [reply postgres_stats.c:70]: Yep. v10 shows all processes in pg_stat_activity, not only backends.

**Comment by a1exsh** [comment]
:+1:

**Pull request merged**